NewDawn/NewDawn/ViewControllers/SettingPageViewControllers/SettingPageViewController.swift

```diff
@@ -20,15 +20,48 @@ class SettingPageViewController: UIViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
         ImageUtil.polishCircularImageView(imageView: profileImage)
+        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(self.imageTapped(gesture:)))// add it to the image view;
+        profileImage.addGestureRecognizer(tapGesture)
+        // make sure imageView can be interacted with by user
+        profileImage.isUserInteractionEnabled = true
+    
         polishUIButton(button: preferenceButton)
         polishUIButton(button: accountButton)
         polishUIButton(button: helpButton)
         polishUIButton(button: settingButton)
-        let dataPath = (NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)[0] as NSString).appendingPathComponent("PersonalImages/0.jpeg")
-        let img = UIImage(contentsOfFile: dataPath)
-        profileImage.image = img
+        
+        LoginUserUtil.fetchLoginUserProfile() {
+            user_profile in
+            if user_profile != nil {
+                if user_profile?.mainImages.isEmpty ?? nil != nil {
```

Comment:
Can you explain a bit what does it mean?
`if user_profile?.mainImages.isEmpty ?? nil != nil `

Reply:
Yeah this looks weird. Just change it to a readable form